docs/data-process.rst

```diff
@@ -355,6 +355,47 @@ YP_TIME_REF=whatever``, then you will get the value ``whatever`` instead of the
 reference time.
 
 
+Cast Value Variable Substitution
+--------------------------------
+
+For non-string scalar values, i.e. integers, floats and booleans, the YAML
+processor utility supports casting the value to the correct type before using
+it for substitution:
+
+``${NAME.int}``
+    Cast value of ``NAME`` to an integer.
+
+``${NAME.float}``
+    Cast value of ``NAME`` to a float.
+
+``${NAME.bool}``
+    Cast value of ``NAME`` to a boolean.
+
+For example, suppose we have ``main.yaml``:
+
+.. code-block:: yaml
+
+   version: ${ITEM_VERSION.int}
+   speed: ${ITEM_SPEED.float}
+
+Running
+:program:`yp-data -D ITEM_VERSION=4 -D ITEM_SPEED=3.14 main.yaml <yp-data>`
```

Comment:
Please provide more good examples and corresponding behaviour you would expect. e.g. what happens if you pass ITEM_VERSION=4.0 or 3.9, or 4.1 to a type of int. What happens if you pass 4 to a type of float. Maybe some esoteric float examples to demonstrate everything supported by python float casting is supported? (or just state this)

Should we support all numeric types: i.e. complex too?
https://docs.python.org/3/library/stdtypes.html#numeric-types-int-float-complex

Reply:
If there is a use case. I have not seen use of complex numbers in our code yet.

Reply:
In modern Python versions, if you try to cast `4.1` to an `int`, you get a `ValueError`. Passing `4` to a `float`, you get `4.0`. This is probably sensible to your implementation of Python. I have added a statement in the doc to say this.